app/models/internal/impl/PagerDutyNotificationEntry.java

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2018 Smartsheet.com
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package models.internal.impl;
+
+import akka.actor.ActorSystem;
+import akka.http.javadsl.Http;
+import akka.http.javadsl.model.ContentTypes;
+import akka.http.javadsl.model.HttpEntities;
+import akka.http.javadsl.model.HttpRequest;
+import com.arpnetworking.commons.builder.OvalBuilder;
+import com.arpnetworking.mql.grammar.AlertTrigger;
+import com.arpnetworking.steno.Logger;
+import com.arpnetworking.steno.LoggerFactory;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.google.inject.Injector;
+import com.typesafe.config.Config;
+import models.internal.Alert;
+import models.internal.NotificationEntry;
+
+import java.net.URI;
+import java.util.Objects;
+import java.util.concurrent.CompletionStage;
+import java.util.stream.Collectors;
+
+/**
+ * Internal model representing a pagerduty notification entry.
+ * These do not have any internal state and act as singletons when creating a NotificationGroup.
```

Comment:
Do we want these to be singletons, or does it seem reasonable to support multiple Pager Duty service key's in a metrics-portal instance?

Reply:
We are definitely going to need to support multiple service keys

Reply:
OK, I'll take a look at this...

Reply:
This one is a bit tricky. I was intending to configure the available endpoints:
pagerDuty = {
  integrations = [
    {
      id = "Test"
      uri = "https://events.pagerduty.com/generic/2010-04-15/create_event.json"
      serviceKey = "abc123456"
    },
    {
      id = "Dummy"
      uri = "https://events.pagerduty.com/generic/2010-04-15/create_event.foo"
      serviceKey = "123qrgqetrhqefwegqweg45"
    },
    {
      id = "Dummy2"
      uri = "https://events.pagerduty.com/generic/2010-04-15/create_event.baz"
      serviceKey = "12345dfefwefwefwefwe"
    }
  ]
}

and to present a dropdown of the ids when the user is adding a pagerduty endpoint - this seems like the friendliest approach. But I'm not seeing a clean way to pass these is strings back to the frontend and making them available to EditNotificationGroupViewModel. I'll continue to investigate (I'm new to much of this technology) but you might have an obvious idea...